modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -3983,14 +3986,65 @@ private CompletableFuture<?> processOperationRequestWithTxRwCounter(
             }
         }
 
-        return processOperationRequest(senderId, request, isPrimary, opStartTsIfDirectRo, leaseStartTime)
-                .whenComplete((unused, throwable) -> {
-                    if (request instanceof ReadWriteReplicaRequest) {
-                        txRwOperationTracker.decrementOperationCount(
-                                rwTxActiveCatalogVersion(catalogService, (ReadWriteReplicaRequest) request)
-                        );
-                    }
-                });
+        UUID txIdLockingLwm = tryToLockLwmIfNeeded(request, opStartTsIfDirectRo);
+
+        try {
+            return processOperationRequest(senderId, request, isPrimary, opStartTsIfDirectRo, leaseStartTime)
+                    .whenComplete((unused, throwable) -> {
+                        unlockLwmIfNeeded(txIdLockingLwm, request);
```

Comment:
I don't like the fact that we detect whether we need to do action on different abstraction levels.
`unlockLwmIfNeeded` hides `if (request instanceof ReadOnlyDirectReplicaRequest) `
but `if (request instanceof ReadWriteReplicaRequest)` is right in front of us. We should either hide both or expose both. In order to have symmetry with tryToLockLwmIfNeeded, I'd prefer hiding both in given case.